connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -763,6 +789,30 @@ private boolean startTask(ConnectorTaskId taskId) {
         );
     }
 
+    private Callable<Void> getTaskCallable(final ConnectorTaskId taskId, final TaskStatus.State targetState) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (targetState) {
+                    case RUNNING:
+                        try {
+                            startTask(taskId);
```

Comment:
> Question: How about we retire the executor in the Worker. It's only used for starting tasks.

That executor is used for all the work tasks do, not just starting (`WorkerTask` is a `Runnable`)